pkg/module/sync/runner.go

```diff
@@ -142,25 +131,31 @@ func (r *Runner) patchManifest(ctx context.Context, owner string, manifestObj *v
 	return nil
 }
 
-func (r *Runner) updateAvailableManifestSpec(ctx context.Context, manifestObj *v1beta2.Manifest) error {
-	manifestInCluster := &v1beta2.Manifest{}
+func NeedToUpdate(kymaStatus v1beta2.KymaStatus, latestManifest *v1beta2.Manifest) bool {
+	moduleStatus := getModuleStatus(kymaStatus, latestManifest.GetName())
 
-	if err := r.Get(ctx, client.ObjectKey{Namespace: manifestObj.GetNamespace(), Name: manifestObj.GetName()},
-		manifestInCluster); err != nil {
-		return fmt.Errorf("error get manifest %s: %w", client.ObjectKeyFromObject(manifestObj), err)
+	if moduleStatus == nil {
+		return true
 	}
-	if !needToUpdate(manifestInCluster, manifestObj) {
-		return nil
-	}
-	manifestInCluster.Spec = manifestObj.Spec
-	if err := r.Update(ctx, manifestInCluster); err != nil {
-		return fmt.Errorf("error update manifest %s: %w", client.ObjectKeyFromObject(manifestObj), err)
+
+	return moduleStatus.Version != latestManifest.Spec.Version ||
+		moduleStatus.Channel != getChannelLabel(latestManifest)
+}
+
+func getModuleStatus(status v1beta2.KymaStatus, moduleName string) *v1beta2.ModuleStatus {
+	for idx := range status.Modules {
+		if status.Modules[idx].Name == moduleName {
+			return &status.Modules[idx]
+		}
 	}
 	return nil
 }
```

Comment:
should this rather be a method on the `KymaStatus` type? Something like: `func(status KymaStatus) GetModuleStatus(name string) (ModuleStatus, err)`?

Reply:
Hmmm I don't think it makes a difference here TBH. Why would you prefer it like this?

Reply:
hmm makes sense. But my point was that it is not used anywhere else so I just had it as a helper method here. But it makes sense to have it on the KymaStatus yes